src/equations/compressible_euler_multicomponent_2d.jl

```diff
@@ -794,13 +797,13 @@ end
 
     rho = density(u, equations)
     gamma = totalgamma(u, equations)
-    rho_times_p = (gamma - 1) * (rho * rho_e - 0.5 * (rho_v1^2 + rho_v2^2))
+    rho_times_p = (gamma - 1) * (rho * rho_e - 0.5f0 * (rho_v1^2 + rho_v2^2))
 
     return rho_times_p
 end
 
 @inline function density(u, equations::CompressibleEulerMulticomponentEquations2D)
-    rho = zero(u[1])
+    rho = 0
```

Comment:
I was also thinking that maybe it is better to use `zero` but the problem is we have adopted using other integers like 2, 3, ... as its original form instead of something like `RealT(2)`, `RealT(3)`, and they also introduce type instability. Why do we only focus on fixing 0 and 1? I mean if we really care about type instability then all integers should be changed.

Reply:
Or we can only make 0/1 to `zero`/`one` as they appear more frequently.

Reply:
That this kind of loops introduces a "real" type instability that impacts performance significantly is kind of mentioned in https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-changing-the-type-of-a-variable. In older versions of Julia, you would have seen a significant impact on performance. Recent versions of Julia fix this since they can analyze that `rho` will only have two possible types, so it compiles an optimized version ("union splitting").

Reply:
Good! You're always the best for answering my questions.